index.js

```diff
@@ -109,9 +109,8 @@ function parseGitUrl (giturl) {
   if (!matched) {
     var legacy = url.parse(giturl)
     if (legacy.auth) {
-      var whatwg = new url.URL(giturl)
-      legacy.auth = whatwg.username || ''
-      if (whatwg.password) legacy.auth += ':' + whatwg.password
+      // Replace the url decoded username:password with the url encoded username:password from the original url
+      legacy.auth = giturl.match(new RegExp('^[^/]+//([^@]+)@'))[1]
```

Comment:
I originally had more checks here for URLs that did not have a protocol or slashes in it, and then I realized I couldn't test that because a URL without a protocol or slashes will never satisfy `legacy.auth` being truthy, so I removed them.